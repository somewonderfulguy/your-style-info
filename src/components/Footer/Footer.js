import React from 'react'

import FooterNavigation from './FooterNavigation'
import LangSelector from '../LangSelector'
import SocialMediaIcons from '../SocialMediaIcons'
import styles from './Footer.module.css'

const Footer = () => (
  <footer className={styles.footer}>
    <div className={styles.layout}>
      <FooterNavigation />
      <div className={styles.iconsContainer}>
        <div className={styles.langContainer}>
          <LangSelector color="gray" showAbove />
        </div>
        <SocialMediaIcons />
      </div>
    </div>
  </footer>
)

export default Footer