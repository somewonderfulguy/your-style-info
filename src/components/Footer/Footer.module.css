.footer {
  background-color: #000;
  color: #919191;
  font-size: 13px;
}

.layout {
  max-width: 1300px;
  margin: 0 auto;
  padding: 50px 100px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

/* tablet */
@media(max-width: 1024px) {
  .layout {
    padding: 50px 75px;
  }
}

/* mobile */
@media(max-width: 768px) {
  .layout {
    padding: 25px;
    flex-direction: column;
    align-items: start;
  }
  .iconsContainer {
    align-self: flex-end;
    margin-top: 20px;
  }
}