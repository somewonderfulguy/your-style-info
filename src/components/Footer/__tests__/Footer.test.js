import React from 'react'
import {render} from '@testing-library/react'

import Footer from '..'

jest.mock('components/DarkThemeSwitcher', () => () => 'MockDarkThemeSwitcher')
jest.mock('components/LangSelector', () => () => 'MockLangSelector')
jest.mock('components//SocialMediaIcons', () => () => 'MockSocialMediaIcons')
jest.mock('components/Footer/FooterNavigation', () => () => 'MockFooterNavigation')

test('should match snapshot', () => {
  const {asFragment} = render(<Footer />)
  expect(asFragment()).toMatchSnapshot()
})