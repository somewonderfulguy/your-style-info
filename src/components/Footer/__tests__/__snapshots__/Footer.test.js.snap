// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="layout"
    >
      MockFooterNavigation
      <div
        class="iconsContainer"
      >
        <div
          class="langContainer"
        >
          MockLangSelector
        </div>
        MockSocialMediaIcons
        <div
          class="themeSwitcherContainer"
        >
          MockDarkThemeSwitcher
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
